File: lab3/shapes/src/Rectangle.java
public class Rectangle implements Shape {
    private int width;
    private int height;
    public Rectangle(int widthIn, int heightIn)
    {
        width = widthIn;
        height = heightIn;
    }
    public String getShapeType()
    {
        return "Rectangle";
    }
    public int getPerimeter()
    {
        return 2 * width + 2 * height;
    }
    public int getArea()
    {
        return width * height;
    }
}

File: lab5-python/shapes/rectangle.py
from shape import Shape

class Rectangle(Shape):
    def __init__(self, name, width, height):
        super().__init__(name)
        self._width = width
        self._height = height
    
    def get_shape_type(self):
        return "Rectangle"

    def get_perimeter(self):
        return self._width * 2 + self._height * 2
    
    def get_area(self):
        return self._width * self._height
